src/js/background.js

```diff
@@ -532,16 +530,22 @@ Badger.prototype = {
    */
   updateBadge: function(tabId){
     if (!this.showCounter()){
-      chrome.browserAction.setBadgeText({tabId: tabId, text: ""});
+      if(chrome.browserAction.setBadgeText){
+        chrome.browserAction.setBadgeText({tabId: tabId, text: ""});
+      }
       return;
     }
     var numBlocked = this.blockedTrackerCount(tabId);
-    if(numBlocked === 0){
-      chrome.browserAction.setBadgeBackgroundColor({tabId: tabId, color: "#00cc00"});
-    } else {
-      chrome.browserAction.setBadgeBackgroundColor({tabId: tabId, color: "#cc0000"});
+    if(chrome.browserAction.setBadgeBackgroundColor){
+      if(numBlocked === 0){
+        chrome.browserAction.setBadgeBackgroundColor({tabId: tabId, color: "#00cc00"});
+      } else {
+        chrome.browserAction.setBadgeBackgroundColor({tabId: tabId, color: "#cc0000"});
+      }
+    }
+    if(chrome.browserAction.setBadgeText){
```

Comment:
re: way to determine if it is mobile

I think we should use feature existence as a way to determine if its a mobile browser, so we could do something like:

```javascript
let isMobile = !(chrome.browserAction.getPopup && chrome.browserAction.setBadgeText && etc);
```

This also means that we won't have to consider edge cases where some of these API's may exist while others do not.

The downside is we won't immediately get new features if these api's are added. But I think think that is okay. And when they are added all we have to do is remove them from the `isMobile` check.